server/src/main/java/org/opensearch/cluster/metadata/IndexNameExpressionResolver.java

```diff
@@ -164,6 +175,10 @@ public String[] concreteIndexNames(ClusterState state, IndicesOptions options, I
         return concreteIndexNames(context, request.indices());
     }
 
+    public List<String> concreteSystemIndices(String... concreteIndices) {
+        return Arrays.stream(concreteIndices).filter(idx -> Regex.simpleMatch(systemIndices, idx)).collect(Collectors.toList());
```

Comment:
Do we care if system index is closed or not?

Reply:
We do. System indices should not be closed. 

I'm glad you brought that up because I see a reference in the security plugin to a non-existent action [`indices:admin/freeze*`](https://github.com/opensearch-project/security/blob/main/src/main/java/org/opensearch/security/privileges/SecurityIndexAccessEvaluator.java#L117C52-L117C72). I'm not aware of the history, but this sounds similar to a close operation which security explicitly denies for system indices.

Reply:
@reta I'm opting not to address this and have this new method act as a filter to the caller to determine if any of the index expressions passed to this method match a registered system index (pattern).